src/pages/VideoPlayer/VideoPlayer.page.js

```diff
@@ -0,0 +1,63 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+import MainPane from '../../components/MainPane';
+import { VideoPlayerContainer } from './VideoPlayer.page.styled';
+import VideoContext from '../../state/videoContext';
+import VideoDetails from '../../components/VideoDetails';
+import { useAuth } from '../../providers/Auth';
+import Button from 'react-bootstrap/Button';
+
+function VideoPlayer() {
+  const { id } = useParams();
+  const { state, dispatch } = useContext(VideoContext);
+  const { currentVideo } = state;
+  const { favoriteVideos } = state;
+  const favorites = favoriteVideos
+    ? favoriteVideos
+    : JSON.parse(localStorage.getItem('favorites'));
+  const video = currentVideo
+    ? currentVideo
+    : JSON.parse(localStorage.getItem('videos'))[id];
+
+  const { authenticated } = useAuth();
+
+  const handleAddFav = (video) => () => {
+    dispatch({ type: 'ADD_FAVORITE', payload: video });
+  };
+
+  const handleRemoveFav = (video) => () => {
+    dispatch({ type: 'REMOVE_FAVORITE', payload: video });
+  };
+
+  function isVideoOnFavorites(video) {
+    if (
+      favorites &&
+      favorites.some((favorite) => favorite.id.videoId === video.id.videoId)
+    ) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  return (
+    <MainPane>
```

Comment:
oh got it, thanks to this component I was able to find the button to add a fav video, however, this seems to be hidden at the bottom of the page, probably you can fix it by playing with styles